cloudwatch_logs_for_cdm.tf

```diff
@@ -0,0 +1,52 @@
+# ------------------------------------------------------------------------------
+# Provision resources to send CDM CloudWatch log events to a Firehose stream
+# that will deliver them to the CDM S3 bucket, where they will be ingested into
+# the CDM system.
+# ------------------------------------------------------------------------------
+
+# Create the Firehose delivery stream that will receive the CloudWatch log
+# events and send them to the CDM S3 bucket.
+resource "aws_kinesis_firehose_delivery_stream" "cloudwatch_logs" {
+  provider = aws.sharedservicesprovisionaccount
+
+  destination = "extended_s3"
+
+  extended_s3_configuration {
+    bucket_arn = aws_s3_bucket.cloudwatch.arn
+    role_arn   = aws_iam_role.firehose_to_s3.arn
+
+    # Decompress each item that passes through the delivery stream.  The
+    # CloudWatch subscription filter automatically GZIPs the log events before
+    # sending them to the Firehose delivery stream, but we want the data to be
+    # decompressed before it is written to the S3 bucket so that Splunk can
+    # ingest it properly.  For more information, see:
+    # https://docs.aws.amazon.com/firehose/latest/dev/writing-with-cloudwatch-logs.html
+    processing_configuration {
+      enabled = "true"
+
+      processors {
+        type = "Decompression"
+
+        parameters {
+          parameter_name  = "CompressionFormat"
+          parameter_value = "GZIP"
+        }
+      }
+    }
+  }
+
+  name = var.firehose_delivery_stream_name
+}
+
+# Create CloudWatch subscription filters to send events from CDM CloudWatch
+# log groups to the Firehose delivery stream
+resource "aws_cloudwatch_log_subscription_filter" "cdm" {
+  provider = aws.sharedservicesprovisionaccount
+  for_each = toset(var.cloudwatch_policy_log_groups)
+
+  destination_arn = aws_kinesis_firehose_delivery_stream.cloudwatch_logs.arn
+  filter_pattern  = ""
```

Comment:
Might be worth mentioning in a comment that this matches everything.

Reply:
Good thinking - I expanded the comment in https://github.com/cisagov/cool-sharedservices-cdm/pull/46/commits/762a214980a3dc92e3ac993a683ada9d3a91c585.